deepdrivemd/agents/stream/config.py

```diff
@@ -0,0 +1,73 @@
+from typing import List, Tuple
+from deepdrivemd.config import AgentTaskConfig
+from pathlib import Path
+
+
+class OutlierDetectionConfig(AgentTaskConfig):
+    """Outlier detection algorithm configuration."""
+
+    # top aggregation directory
+    agg_dir: Path = (
```

Comment:
Should be:

```
agg_dir: Path = Path(
        "/usr/workspace/cv_ddmd/yakushin/Integration1/Outputs/1/aggregation_runs"
    )
```

Currently the type is wrong. However, I think we should remove this hardcoded path from the config since it is lassen specific and this code should be as general as possible.

Reply:
All these config.py files are extremely ugly, redundant and error prone. They exist only for the benefit of pydantic to do some input data type error checking? I suggest that we simply import the corresponding class from generate.py: this way there is no redundancy and no possibility to introduce inconsistency between yaml and config.py. generate.py is necessary in any case since editing yaml by hand is not for humans and due to repetition of some parameters in different components it is again easy to introduce inconsistency if yaml is edited manually.

Reply:
The data classes in generate.py function the same way as pydantic classes. In the simplest form a pydantic class is just a data class like you have in generate.py. Can I suggest that instead of generate.py making additional data classes, that it use the pydantic classes as follows:

```

cfg = ExperimentConfig(
        title="COVID-19 - Workflow2",
        resource="ornl.summit",
        queue="batch",
        schema_="local",
        project="MED110",
        walltime_min=360,
        cpus_per_node=42,
        hardware_threads_per_cpu=4,
        gpus_per_node=6,
        max_iteration=4,
        experiment_directory="/path/to/experiment",
        node_local_path=None,
        molecular_dynamics_stage=MolecularDynamicsStageConfig(),
        aggregation_stage=AggregationStageConfig(),
        machine_learning_stage=MachineLearningStageConfig(),
        model_selection_stage=ModelSelectionStageConfig(),
        agent_stage=AgentStageConfig(),
    )
    
 cfg.molecular_dynamics_stage.task_config.pre_exec = my_pre_exec
 # Other updates ...
 
 cfg.dump_yaml("config.yaml")
```

Here I uses the file-based DDMD config, but the same logic holds for `StreamingExperimentConfig`. This would reduce a lot of confusion and avoid the need to have two separate data classes. I like keeping the task level configs with the task level logic to encapsulate the parameters with the code. Also we need the ability to quickly implement different algorithms and associated data classes which seamlessly interface with the framework.

Reply:
Done

Reply:
I meant done fixing the path, not the big change.

Reply:
I do not understand what you are suggesting.

I was thinking of importing the corresponding class from generate.py into config.py (the classes in generate.py would need data types specified), inheriting from it in order to introduce various pydantic validate statements. If we do it it this way, it does not matter how the classes are named in generate.py since those will be inherited into more pydantic friendly classes which can be named they way they are currently named. 

A user of generate.py does not need to know anything about pydantic, he just needs the simplest way to generate error-free yaml, any kind of constraints coming from pydantic would only make things more confusing and error-prone.

Reply:
generate.py should be completely self-contained and independent from anything to be user friendly.